packages/graphback-core/src/plugin/getSelectedFieldsFromResolverInfo.ts

```diff
@@ -16,23 +22,28 @@ export const getSelectedFieldsFromResolverInfo = (info: GraphQLResolveInfo, mode
       acc: Set<string>,
       entry: [string, unknown]
     ) => {
+      let selectedField: string;
 
       if (typeof entry[1] !== 'object') {
-        acc.add(entry[0]);
+        selectedField = entry[0];
       } else {
         // it is a nested object, check if it is a relationship to add the appropriate foreign key field
         const foundRelationship = model.relationships.find((relationship: FieldRelationshipMetadata) => relationship.ownerField.name === entry[0]);
         if (foundRelationship) {
           if (foundRelationship.kind !== "oneToMany") {
-            acc.add(foundRelationship.relationForeignKey);
+            selectedField = foundRelationship.relationForeignKey;
           } else {
-            acc.add(model.primaryKey);
+            selectedField = model.primaryKey;
           }
         } else {
-          acc.add(entry[0])
+          selectedField = entry[0];
         }
       }
 
+      if (selectedField && !INTERNAL_NON_USER_FIELDS.includes(selectedField)) {
+        acc.add(selectedField);
+      }
```

Comment:
I think the most reliable way is having the model fields information in [`ModelDefinition`](https://github.com/aerogear/graphback/blob/master/packages/graphback-core/src/plugin/ModelDefinition.ts#L8). This will also include foreign keys / relationships join key.  Afterwards we can safely ignore any non model field. 

I'll add this info in https://github.com/aerogear/graphback/issues/1675 so that we parse the fields info there too.